packages/astro/src/core/build/plugins/plugin-pages.ts

```diff
@@ -44,8 +44,8 @@ function vitePluginPages(opts: StaticBuildOptions, internals: BuildInternals): V
 				for (const pageData of pageDatas) {
 					const resolvedPage = await this.resolve(pageData.moduleSpecifier);
 					if (resolvedPage) {
-						imports.push(`const page = () => import(${JSON.stringify(pageData.moduleSpecifier)});`);
-						exports.push(`export { page }`);
+						imports.push(`import * as _page from ${JSON.stringify(pageData.moduleSpecifier)};`);
+						exports.push(`export const page = () => _page`);
```

Comment:
I reverted this part due to an edgecase at in the [db no-apptoken test](https://github.com/withastro/astro/blob/135a93bce8e0ee663901a4df73bcbece442585a1/packages/db/test/ssr-no-apptoken.test.js#L27).

The problem is here: 

https://github.com/withastro/astro/blob/135a93bce8e0ee663901a4df73bcbece442585a1/packages/astro/src/core/app/index.ts#L312-L324

The nested dynamic import shifts the error to only throw on the second await. If we remove the nested dynamic import, it will then throw on the first await, which is too early to catch any errors.

The fix could be simply moving the first await inside the try..catch block, but I want to prevent unnecessary changes for now and tackle that separately. It doesn't degrade the prerendering chunk handling.

Reply:
I am more than happy to send a PR to change this